### PR TITLE
fix: Send identify event.

### PR DESCRIPTION
**Pull request by yusinto**

### CODE REVIEW

File: packages/shared/sdk-client/src/LDClientImpl.ts [hunk 1]
```diff
@@ -168,20 +168,8 @@ export default class LDClientImpl implements LDClient {
...
-
-        if (changedKeys.length > 0) {
-          this.logger.debug(`Emitting changes from PUT: ${changedKeys}`);
-          // emitting change resolves identify
-          this.emitter.emit('change', context, changedKeys);
-        } else {
-          // manually resolve identify
-          this.logger.debug('Not emitting changes from PUT');
-          identifyResolve();
-        }
```
Comment (yusinto) [inline packages/shared/sdk-client/src/LDClientImpl.ts:184]: Dry'd this to `onIdentifyResolve`.

File: packages/shared/sdk-client/src/LDClientImpl.ts [hunk 2]
```diff
@@ -297,17 +284,14 @@ export default class LDClientImpl implements LDClient {
       return Promise.reject(error);
     }
 
-    const { identifyPromise, identifyResolve } = this.createPromiseWithListeners();
+    this.eventProcessor?.sendEvent(this.eventFactoryDefault.identifyEvent(checkedContext));
```
Comment (yusinto) [inline packages/shared/sdk-client/src/LDClientImpl.ts:287]: This was the missing piece to send `identify` event.

File: packages/shared/sdk-client/src/events/createEventProcessor.ts [hunk 3]
```diff
@@ -1,5 +1,4 @@
 import { ClientContext, internal, Platform } from '@launchdarkly/js-sdk-common';
-import { EventProcessor } from '@launchdarkly/js-sdk-common/dist/internal';
```
Comment (yusinto) [inline packages/shared/sdk-client/src/events/createEventProcessor.ts:2]: This is an unrelated bug. I'm not sure why this wasn't caught by the linter.

File: packages/shared/sdk-client/src/LDClientImpl.ts [hunk 4]
```diff
@@ -263,7 +251,6 @@ export default class LDClientImpl implements LDClient {
 
       this.identifyChangeListener = (c: LDContext, changedKeys: string[]) => {
         this.logger.debug(`change: context: ${JSON.stringify(c)}, flags: ${changedKeys}`);
-        resolve();
```
Comment (yusinto) [inline packages/shared/sdk-client/src/LDClientImpl.ts:266]: Resolve not gets explicitly called when needed, rather than here. I think this change makes the code more readable.

File: packages/shared/sdk-client/src/LDClientImpl.ts [hunk 5]
```diff
@@ -168,20 +168,8 @@ export default class LDClientImpl implements LDClient {
       deserializeData: JSON.parse,
       processJson: async (dataJson: Flags) => {
         this.logger.debug(`Streamer PUT: ${Object.keys(dataJson)}`);
-        const changedKeys = calculateFlagChanges(this.flags, dataJson);
-        this.context = context;
-        this.flags = dataJson;
+        this.onIdentifyResolve(identifyResolve, dataJson, context, 'streamer PUT');
```
Comment (kinyoklion) [inline packages/shared/sdk-client/src/LDClientImpl.ts:171]: This looks like it needs to be after:
`await this.platform.storage?.set(canonicalKey, JSON.stringify(this.flags));`

If it is before, and you await identify and then evaluate a flag, you would get old values instead of the new values.

**Review by kinyoklion — APPROVED** [review]
